### PR TITLE
BaSP-MR43: Members table, add, delete and edit

### PR DESCRIPTION
**Pull request by MatiasSauvanet**
Create a table, add, edit and delete with react to delete Members in the server. Still miss the css to made it pretty.

### CODE REVIEW

File: package-lock.json
Comment (DelfiLT) [inline package-lock.json:1]: remove package-lock.json from your changes

File: src/Components/Members/index.js [hunk 1]
```diff
@@ -3,22 +3,396 @@ import styles from './members.module.css';
...
       });
+
+      const { message, data, error } = await newMembers.json();
+      alert(message);
+
+      if (!error) {
+        setMembers([...members, data]);
+      }
+    } catch (error) {
+      console.log(error);
```
Comment (DelfiLT) [inline src/Components/Members/index.js:126]: Change all of these for change.error

File: src/Components/Members/index.js [hunk 2]
```diff
@@ -3,22 +3,396 @@ import styles from './members.module.css';
...
+      if (confirmDelete) {
+        const apiResponse = await fetch(`${process.env.REACT_APP_API_URL}/api/members/${id}`, {
+          method: 'DELETE'
+        });
+        const data = await apiResponse.json();
+        alert(data.message);
+        setMembers([...members.filter((members) => members._id !== id)]);
+      }
+    } catch (error) {
+      console.error(error);
```
Comment (DelfiLT) [inline src/Components/Members/index.js:142]: Same here

File: src/Components/Members/index.js [hunk 3]
```diff
@@ -3,22 +3,396 @@ import styles from './members.module.css';
...
+    }
+  };
+
+  const getMembers = async () => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/members`);
+      const data = await response.json();
+      setMembers(data.data);
+    } catch (error) {
+      console.error(error);
```
Comment (DelfiLT) [inline src/Components/Members/index.js:187]: Same here

File: src/Components/Members/index.js [hunk 4]
```diff
@@ -3,22 +3,396 @@ import styles from './members.module.css';
...
+    dni,
+    phone,
+    email,
+    city,
+    birthDay,
+    postalCode,
+    isActive,
+    membership
+  });
+  // const [memberFormValue, setMemberFormValue] = useState({
```
Comment (DelfiLT) [inline src/Components/Members/index.js:28]: In the final commit remember to delete the comments

File: src/Components/Members/index.js [hunk 5]
```diff
@@ -3,22 +3,396 @@ import styles from './members.module.css';
...
+      const { message, data, error } = await res.json();
+      alert(message);
+
+      if (!error) {
+        const actualMembers = [...members];
+        actualMembers[adminIndex] = data;
+        setMembers(actualMembers);
+      }
+    } catch (error) {
+      console.error;
```
Comment (SantiH97) [inline src/Components/Members/index.js:166]: same here

File: src/Components/Members/index.js [hunk 6]
```diff
@@ -3,22 +3,396 @@ import styles from './members.module.css';
...
+  };
+
+  const getByIdMember = async (id) => {
+    try {
+      const res = await fetch(`${process.env.REACT_APP_API_URL}/api/members${id}`);
+      const { data } = await res.json();
+
+      return data;
+    } catch (error) {
+      console.error;
```
Comment (SantiH97) [inline src/Components/Members/index.js:177]: same here

**Review by SantiH97 — DISMISSED** [review]
Change those and you should be good to go

File: src/Components/Members/index.js [hunk 7]
```diff
@@ -3,22 +3,393 @@ import styles from './members.module.css';
...
+      cityAdd,
+      birthDayAdd,
+      postalCodeAdd,
+      isActiveAdd,
+      membershipAdd
+    });
+  };
+
+  const addMember = async (member) => {
+    console.log(member);
```
Comment (SantiH97) [inline src/Components/Members/index.js:106]: Delete unnecessary console.log

File: src/Components/Members/index.js (hunk 1 above)
Comment (Dannylez) [reply src/Components/Members/index.js]: Do you mean "console.error"?

**Review by Dannylez — CHANGES_REQUESTED** [review]
remove package-lock.json from your changes

File: src/Components/Members/index.js (hunk 1 above)
Comment (DelfiLT) [reply src/Components/Members/index.js]: Yes sorry

**Review by DelfiLT — APPROVED** [review]

**Review by Dannylez — APPROVED** [review]

File: src/Components/Members/index.js (hunk 7 above)
Comment (DelfiLT) [inline src/Components/Members/index.js:106]: Delete console.log

**Review by Juanoauil — APPROVED** [review]

**Pull request merged**